scale-tests/run.sh

```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+function usage () {
+  echo 'Usage: ./run.sh \\'
+  echo '         <path to test configuration file> \\'
+  echo '         <test name> \\'
+  echo '         <test S3 bucket> \\'
+  echo '         <test S3 folder> \\'
+  echo '         <path to cluster SSH private key> \\'
+  echo '         <cluster URL> \\'
+  echo '         <DCOS username> \\'
+  echo '         <DCOS password> \\'
+  echo '         <cluster security mode> (optional, defaults to permissive)'
+  echo
+  echo 'Example: ./run.sh \\'
+  echo '           scale-tests/configs/2018-01-01.env \\'
+  echo '           scale-tests-2018-01-01 \\'
+  echo '           infinity-artifacts \\'
+  echo '           scale-tests/2018-01-01 \\'
+  echo '           ~/.ssh/dcos \\'
+  echo '           https://scaletests.mesosphere.io \\'
+  echo '           john \\'
+  echo '           john123 \\'
+  echo '           strict'
+}
+
+if [ "${#}" -lt 8 ]; then
+  echo -e "run.sh needs at least 8 arguments but was given ${#}\\n"
+  usage
+  exit 1
+fi
+
+readonly REQUIREMENTS='git docker maws tee'
+
+for requirement in ${REQUIREMENTS}; do
+  if ! [[ -x $(command -v "${requirement}") ]]; then
+    echo "You need to install '${requirement}' to run this script"
+    exit 1
+  fi
+done
+
+readonly TEST_CONFIG="${1:-}"
+readonly TEST_NAME="${2:-}"
+readonly TEST_S3_BUCKET="${3:-}"
+readonly TEST_S3_FOLDER="${4:-}"
+readonly CLUSTER_SSH_KEY="${5:-}"
+readonly CLUSTER_URL="${6:-}"
+readonly DCOS_USERNAME="${7:-}"
+readonly DCOS_PASSWORD="${8:-}"
+readonly SECURITY="${9:-permissive}"
+
+for file in "${CLUSTER_SSH_KEY}" "${TEST_CONFIG}"; do
+  if ! [[ -s ${file} ]]; then
+    echo "File '${file}' doesn't exist or is empty"
+    exit 1
+  fi
+done
+
+readonly AWS_ACCOUNT='Team 10'
+readonly CLUSTER_SPARK_PACKAGE_REPO='https://universe-converter.mesosphere.com/transform?url=https://infinity-artifacts.s3.amazonaws.com/permanent/spark/assets/scale-testing/stub-universe-spark.json' # Spark with quota support.
+readonly CONTAINER_NAME="${TEST_NAME}"
+readonly CONTAINER_SSH_AGENT_EXPORTS=/tmp/ssh-agent-exports
+readonly CONTAINER_SSH_KEY=/ssh/key
+readonly IMAGE_NAME="mesosphere/dcos-commons:${TEST_NAME}"
+readonly LOG_FILE="${TEST_NAME}.log"
+readonly TEST_DIRECTORY="${TEST_NAME}"
+
+source "${TEST_CONFIG}"
+
+function log {
+  local -r message="${*:-}"
+  echo "$(date "+%Y-%m-%d %H:%M:%S") | ${message}" 2>&1 | tee -a "${LOG_FILE}"
+}
+
+function container_exec () {
+  local -r command="${*}"
+  local -r full_command="
+    . ${CONTAINER_SSH_AGENT_EXPORTS}
+    ssh-add -k ${CONTAINER_SSH_KEY} > /dev/null 2>&1
```

Comment:
Why re-add the key with every command, rather than just at the beginning?

Reply:
Because this is running in a container.

Reply:
Why does it matter? `ssh-agent` keeps running in that container when it's started. It will keep the key loaded for as long as it's running, there is no point re-adding that key a million times.

Reply:
@porridge you're right. The sourcing of the ssh-agent environment variables (line 80, `. ${CONTAINER_SSH_AGENT_EXPORTS}`) is enough to wire the shell instance with the ssh-agent session.

Reply:
@porridge https://github.com/mesosphere/spark-build/pull/348/commits/de5fb303a534d11282f806b3de5f2aaaf56d0166